src/ingestion/collector.py

```diff
@@ -9,73 +12,105 @@
 from .pipeline import Pipeline
 
 
-def flatten(arr):
-    return list(itertools.chain.from_iterable(arr))
+class MonitorType(Enum):
+    SCHEMA = 'schema'
+    TABLE = 'table'
+
+@dataclass
+class Monitor(DataClassDictMixin):
+    type: str
+    definition: Optional[Dict] = None
 
 @dataclass
-class CollectorState:
-    discovered_data: Any = None
-    fetched_data: Any = None
+class CollectorConfiguration(DataClassDictMixin): # This is technically repr of what YAML used to be
+    monitors: List[Monitor]
+
+    @classmethod
+    def default(cls):
+        return cls(monitors=[])
+
+    def filter_by_monitors(self, ddata):
+        pass
 
 @dataclass
 class Collector:
     source: Source
-    filters: List[Any] # pure py function
     pipelines: List[Pipeline]
-    state: CollectorState = CollectorState()
+    configuration: CollectorConfiguration = CollectorConfiguration.default()
 
-    def _create_tasks(self):
+    def _create_tasks(self, discovered_data):
```

Comment:
is it necessary to pass `discovered_data` as a parameter here?

Reply:
here? No - in general? yes